src/BlockchainReader/ClaimController.ts

```diff
@@ -15,55 +17,48 @@ export class ClaimController {
   private readonly db: Db
   private readonly collection: Collection
   private readonly messaging: Messaging
-  private readonly insightHelper: InsightClient
+  private readonly bitcoinCore: BitcoinCore
   private readonly configuration: ClaimControllerConfiguration
 
   constructor(
     @inject('Logger') logger: Pino.Logger,
     @inject('DB') db: Db,
     @inject('Messaging') messaging: Messaging,
-    @inject('InsightHelper') insightHelper: InsightClient,
+    @inject('BitcoinCore') bitcoinCore: BitcoinCore,
     @inject('ClaimControllerConfiguration') configuration: ClaimControllerConfiguration
   ) {
     this.logger = childWithFileName(logger, __filename)
     this.db = db
     this.messaging = messaging
-    this.insightHelper = insightHelper
+    this.bitcoinCore = bitcoinCore
     this.configuration = configuration
     this.collection = this.db.collection('blockchainReader')
   }
 
   async scanBlock(blockHeight: number): Promise<void> {
     const logger = this.logger.child({ method: 'scanBlock' })
 
-    logger.trace({ blockHeight }, 'Retrieving Block Hash...')
+    logger.debug({ blockHeight }, 'Retrieving Block Hash')
 
-    const blockHash = await this.insightHelper.getBlockHash(blockHeight)
+    const blockHash = await this.bitcoinCore.getBlockHash(blockHeight)
 
     logger.trace(
       {
         blockHeight,
         blockHash,
       },
-      'Block Hash retrieved successfully. Retrieving Raw Block...'
+      'Block Hash retrieved successfully. Retrieving Raw Block'
     )
 
-    const block = await this.insightHelper.getBlock(blockHash)
-    const poetTimestamps: ReadonlyArray<PoetTimestamp> = block.transactions
-      .map(getPoetTimestamp)
-      .filter(_ => _)
-      .map(_ => ({
-        ..._,
-        blockHeight,
-        blockHash,
-      }))
-    const transactionIds = block.transactions.map(_ => _.id)
+    const block: Block = await this.bitcoinCore.getBlock(blockHash, GetBlockVerbosity.Transactions)
+
+    const poetAnchors = blockToPoetAnchors(block)
 
-    const matchingPoetTimestamps = poetTimestamps
+    const matchingPoetTimestamps = poetAnchors
```

Comment:
This part can be extracted into pure functions and unit tested. I'll work on it.